### PR TITLE
Reduce CPU usage

### PR DESCRIPTION
**Pull request by Williangalvani**
This reduces the CPU usage on a Pi Zero to around 10% when polling at 10Hz with the simplePingExample.py

### CODE REVIEW

File: brping/ping1d.py [hunk 1]
```diff
@@ -70,9 +73,11 @@ def __init__(self, device_name, baudrate=115200):
     # @return A new PingMessage: as soon as a message is parsed (there may be data remaining in the buffer to be parsed, thus requiring subsequent calls to read())
     # @return None: if the buffer is empty and no message has been parsed
     def read(self):
-        while self.iodev.in_waiting:
-            b = self.iodev.read()
+        bytes = self.iodev.read(self.iodev.in_waiting)
+        self._input_buffer.extendleft(bytes)
 
+        while len(self._input_buffer):
+            b = chr(self._input_buffer.pop())
```
Comment (patrickelectric) [inline brping/ping1d.py]: Is `chr` necessary if we cast it to `ord` in the next line ?

File: brping/ping1d.py [hunk 2]
```diff
@@ -11,6 +11,7 @@
 from brping import pingmessage
 import serial
 import time
+from collections import deque
```
Comment (patrickelectric) [inline brping/ping1d.py]: Should collections be before brping import ? For import/alphabetic reasons.

File: brping/ping1d.py (hunk 1 above)
Comment (Williangalvani) [reply brping/ping1d.py]: good catch, that simplifies things.

File: brping/ping1d.py (hunk 2 above)
Comment (Williangalvani) [reply brping/ping1d.py]: [Pep8 suggests something else](https://www.python.org/dev/peps/pep-0008/#imports). And by that we are both wrong, I'll re-organize things

**Pull request merged**